src/tests/curry.test.ts

```diff
@@ -151,15 +152,59 @@ describe('curry module', () => {
   })
 
   describe('debounce function', () => {
+    let func: DebounceFunction<any>
+    const mockFunc = jest.fn()
+    const runFunc3Times = () => _.iterate(3, func, undefined)
+
+    beforeEach(() => {
+      func = _.debounce({ delay: 600 }, mockFunc)
+
+    })
+    
+    afterEach(() => {
+      jest.clearAllMocks()
+    })
+    
     test('only executes once when called rapidly', async () => {
-      let calls = 0
-      const func = _.debounce({ delay: 600 }, () => calls++)
-      func()
+      runFunc3Times()
+      expect(mockFunc).toHaveBeenCalledTimes(0)
+      await _.sleep(610)
+      expect(mockFunc).toHaveBeenCalledTimes(1)
+    })
+
+    test("after calling cancel there should be no more debounce", () => { 
```

Comment:
```suggestion
    test("does not debounce after cancel is called", () => { 
```
making it a bit more readable in test output. With the describe it will now be `debounce function > does not debounce after cancel is called`